src/rgw/rgw_sal_motr.cc

```diff
@@ -1893,6 +1897,11 @@ int MotrObject::read_mobj(const DoutPrefixProvider* dpp, int64_t off, int64_t en
     left -= actual;
     // Read from Motr.
     op = nullptr;
+    if( start > block_start_off )
```

Comment:
As discussed let's chnage the logic a bit to avoid any negative values in bloff - 

```
block_start_off = 0;
if( start > block_start_off + bs )
        block_start_off += bs;
	continue;
}
...
bloff = start - block_start_off
```

Reply:
bs may get changed at line 1884, hence logic suggested by Saurabh should be used